docs/diagrams/ApplyCommandActivity.puml

```diff
@@ -1,28 +1,29 @@
 @startuml
 skinparam {
   backgroundColor white
-
   ArrowColor black
   ArrowThickness 2
-
   ActivityBackgroundColor green
   ActivityBorderColor #2980B9
   ActivityFontColor white
 }
-
 start
 :User adds application\n(executes ApplyCommand);
 
-If () then ([Command is invalid])
+if () then ([Command is invalid])
     :Displays message for invalid command format;
+else ([Valid command])
+    :Gets company by index;
 
-    else ([else])
+    :Adds application to the company;
 
-        :Gets company by index;
-        :Adds application to the company;
-        :Display updated list of companies;
+    if () then ([else])
+        :Change company status to "APPLIED";
+    else ([Company status is "APPLIED"])
+    endif
 
-        endif
+endif
+:Display updated company in current list view;
```

Comment:
I think the right branch should end with an `display success message` and then the activity should end after the merge node (without `display updated company in current list view` since the left branch does not necessarily update the company.